src/neighbors.cc

```diff
@@ -1408,13 +1408,17 @@ peerCountMcastPeersStart(void *data)
     const MasterXaction::Pointer mx = new MasterXaction(XactionInitiator::initPeerMcast);
     auto *req = HttpRequest::FromUrlXXX(url, mx);
     assert(req != nullptr);
+    const AccessLogEntry::Pointer ale = new AccessLogEntry;
+    ale->request = req;
+    CodeContext::Reset(ale);
```

Comment:
In this approach, the context is provided by creating ALE for the ICP transaction. Note that providing the ALE for the transaction does not lead to logging because icpLogIcp() ignores LOG_ICP_QUERY requests, i.e., the old logging behavior remains unchanged. On the other hand, the fact that LOG_ICP_QUERY transactions are not logged is at odds with log_icp_queries parameter ('on' by default), controlling whether ICP queries are logged. I doubt we should not address this problem (if it is indeed a problem) in this PR.